### PR TITLE
Beta Support 

### PR DESCRIPTION
**Pull request by finsharp**

### CODE REVIEW

File: ApiDoctor.Console/App.config [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8" ?>
 <configuration>
-    <configSections>
-    </configSections>
-    <startup> 
-        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5" />
-    </startup>
+  <runtime>
+    <gcServer enabled="true"/>
```
Comment (daspek) [inline ApiDoctor.Console/App.config]: why specify the gc type?

File: ApiDoctor.Validation/DocSet.cs [hunk 2]
```diff
@@ -230,11 +229,11 @@ private void LoadTestScenarios()
             var jsonFiles = docSetDir.GetFiles("*.json", SearchOption.AllDirectories);
             foreach (var file in jsonFiles)
             {
-                if (file.DirectoryName.Contains("beta"))
-                {
-                    // hack: skip beta for now.
-                    continue;
-                }
+                //if (file.DirectoryName.Contains("beta"))
```
Comment (daspek) [inline ApiDoctor.Validation/DocSet.cs]: remove instead of comment

File: ApiDoctor.Validation/DocSet.cs [hunk 3]
```diff
@@ -659,15 +658,18 @@ private void ReadDocumentationHierarchy(string path)
                 throw new FileNotFoundException(string.Format("Cannot find documentation. Directory doesn't exist: {0}", sourceFolder.FullName));
             }
 
-            var markdownFileInfos = sourceFolder.GetFiles(DocumentationFileExtension, SearchOption.AllDirectories);
-            var markdownFiles = markdownFileInfos.
-                Where(fi => foldersToSkip.All(folderToSkip => !fi.DirectoryName.Contains(folderToSkip)) && !filesToSkip.Contains(fi.Name)).
-                Select(fi => new DocFile(this.SourceFolderPath, this.RelativePathToFile(fi.FullName), this) { WriteFixesBackToDisk = this.writeFixesBackToDisk });
+            var markdownFileInfos = sourceFolder.GetFiles(DocumentationFileExtension, SearchOption.AllDirectories)
+                .Where(x => !x.Name.Contains("intune")).ToArray();
```
Comment (daspek) [inline ApiDoctor.Validation/DocSet.cs]: is this necessary? what in intune isn't passing? if we must add this, let's not put it in the code. instead, let's add 'pathSubstringsToSkip' (or something) to one of the config files (like schema-config, even though that's not super appropriate) and read it from there. that way only the graph repo will have this configuration

File: ApiDoctor.Validation/Http/HttpParser.cs [hunk 4]
```diff
@@ -23,115 +23,118 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+
+
```
Comment (daspek) [inline ApiDoctor.Validation/Http/HttpParser.cs]: put these under the `namespace` near the other usings

File: ApiDoctor.Validation/Http/HttpParser.cs [hunk 5]
```diff
@@ -23,115 +23,118 @@
...
         /// <returns></returns>
         public HttpRequest ParseHttpRequest(string requestString)
         {
             if (string.IsNullOrWhiteSpace(requestString))
-            {
                 throw new ArgumentException("requestString was empty or whitespace only. Not a valid HTTP request.");
-            }
 
-            StringReader reader = new StringReader(requestString);
+            var firstNewLine = requestString.IndexOf(Environment.NewLine, StringComparison.InvariantCultureIgnoreCase);
```
Comment (daspek) [inline ApiDoctor.Validation/Http/HttpParser.cs]: what's this trying to do? trim any newlines before the request starts?
1) why not just fix the docs? i think that's what i did for v1.
2) why not just call trim()?

File: ApiDoctor.Validation/Http/HttpParser.cs [hunk 6]
```diff
@@ -23,115 +23,118 @@
...
-
-                        if (components.Length >= 3)
-                        {
-                            request.HttpVersion = components[2];
-                        }
-                        else
-                        {
-                            request.HttpVersion = "HTTP/1.1";
-                        }
+                        request.Url = componentsLength > 3 ? line.Replace(request.Method, string.Empty) : components[1];
```
Comment (daspek) [inline ApiDoctor.Validation/Http/HttpParser.cs]: what's this doing?

File: ApiDoctor.Validation/Http/HttpParser.cs [hunk 7]
```diff
@@ -23,115 +23,118 @@
...
+                        request.HttpVersion = components[componentsLength - 1].StartsWith("HTTP/")
+                            ? components[componentsLength - 1]
+                            : "HTTP/1.1";
 
                         mode = ParserMode.Headers;
                         break;
 
                     case ParserMode.Headers:
-                        if (string.IsNullOrEmpty(line))
+                        if (string.IsNullOrEmpty(line) || line.StartsWith("{", true, CultureInfo.InvariantCulture))
```
Comment (daspek) [inline ApiDoctor.Validation/Http/HttpParser.cs]: use StringComparison.Ordinal or OrdinalIgnoreCase when we're matching exact tokens like these (ie not user input)

File: ApiDoctor.Validation/Http/HttpParser.cs [hunk 8]
```diff
@@ -23,115 +23,118 @@
...
-
                         break;
 
                     case ParserMode.Body:
                         var restOfBody = reader.ReadToEnd() ?? string.Empty;
 
                         // normalize line endings to CRLF, which is required for headers, etc.
                         restOfBody = restOfBody.Replace("\r\n", "\n").Replace("\n", "\r\n");
-                        request.Body = string.Concat(line, "\r\n", restOfBody);
+                        request.Body = string.Concat(bodyBrace, line, "\r\n", restOfBody);
```
Comment (daspek) [inline ApiDoctor.Validation/Http/HttpParser.cs]: why did you need to do the bodyBrace stuff?

File: ApiDoctor.Validation/Json/JsonSchema.cs [hunk 9]
```diff
@@ -442,7 +442,7 @@ private PropertyValidationOutcome ValidateProperty(ParameterDefinition inputProp
                         }
                         else if (null == inputProperty.Type.CustomMembers)
                         {
-                            issues.Error(ValidationErrorCode.NoCustomMembersFound, $"Property '{inputProperty.Name}' is of type Custom but has no custom members.");
+                            //issues.Error(ValidationErrorCode.NoCustomMembersFound, $"Property '{inputProperty.Name}' is of type Custom but has no custom members.");
```
Comment (daspek) [inline ApiDoctor.Validation/Json/JsonSchema.cs]: add a comment saying this is temporarily disabled and put a link to the bug number (ideally github issue) assigned to me to investigate it

File: ApiDoctor.Validation/TableSpec/tablespecconverter.cs [hunk 10]
```diff
@@ -84,8 +84,8 @@ public TableDefinition ParseTableSpec(Block tableSpecBlock, Stack<Config.Documen
 
             TableDecoder decoder = new TableDecoder { Type = TableBlockType.Unknown };
 
-            var headerText = headerStack.Peek()?.Title;
-            
+            var headerText = headerStack.Count > 0 ? headerStack.Peek()?.Title : null;
```
Comment (daspek) [inline ApiDoctor.Validation/TableSpec/tablespecconverter.cs]: this looks equivalent to what it was doing before. why the change?

File: ApiDoctor.Console/App.config (hunk 1 above)
Comment (finsharp) [reply ApiDoctor.Console/App.config]: This was an investigation into improving performance. By default it uses workstation gc. I will remove this pending further investigation as it did not bear fruit

File: ApiDoctor.Validation/DocSet.cs (hunk 2 above)
Comment (finsharp) [reply ApiDoctor.Validation/DocSet.cs]: Done

File: ApiDoctor.Validation/DocSet.cs (hunk 3 above)
Comment (finsharp) [reply ApiDoctor.Validation/DocSet.cs]: Good point. Yes intune has a lot of errors, that I cannot fix ( I was fixing the docs manually earlier on). I was informed intune docs are autogenerated by another system.

File: ApiDoctor.Validation/Http/HttpParser.cs (hunk 5 above)
Comment (finsharp) [reply ApiDoctor.Validation/Http/HttpParser.cs]: I realized I was wrong here. I fixed this.

File: ApiDoctor.Validation/TableSpec/tablespecconverter.cs (hunk 10 above)
Comment (finsharp) [reply ApiDoctor.Validation/TableSpec/tablespecconverter.cs]: Would throw an exception  if Peek() was called and the stack was empty.

File: ApiDoctor.Validation/Http/HttpParser.cs (hunk 8 above)
Comment (finsharp) [reply ApiDoctor.Validation/Http/HttpParser.cs]: Corrected

File: ApiDoctor.Validation/Http/HttpParser.cs (hunk 7 above)
Comment (finsharp) [reply ApiDoctor.Validation/Http/HttpParser.cs]: Corrected.

File: ApiDoctor.Validation/Http/HttpParser.cs (hunk 6 above)
Comment (finsharp) [reply ApiDoctor.Validation/Http/HttpParser.cs]: I had tried to handle edge cases arising from doc write mistakes. I decided to abandon this and force them to fix the docs.
